frontend/src/components/GAlert.vue

```diff
@@ -5,69 +5,101 @@ SPDX-License-Identifier: Apache-2.0
 -->
 
 <template>
-  <v-alert :color="color" :tile="tile" :dark="dark" dismissible v-model="alertVisible">
-    <div class="subtitle-1">
-      {{message}}
-      <v-btn dark outlined small v-if="!!detailedMessage" @click="detailedMessageVisible = !detailedMessageVisible">
-        Details
-      </v-btn>
-    </div>
-    <transition name="fade">
-      <div v-if="!!detailedMessageVisible">
-        <code>{{detailedMessage}}</code>
-      </div>
-    </transition>
+  <v-alert
+    class="alertBanner"
+    :type="type"
+    v-model="alertVisible"
+    :color="color"
+    :transition="transition"
+    @input="closeBanner"
+    dismissible
+  >
+    <div v-if="message" class="alert-banner-message" v-html="messageHtml"></div>
+    <slot v-else name="message"></slot>
   </v-alert>
 </template>
 
 <script>
+import { mapActions } from 'vuex'
+import { transformHtml } from '@/utils'
+
+const LOCAL_STORE_ALERT_BANNER_HIDDEN_MESSAGES = 'global/alert-banner/hidden-messages'
+
 export default {
-  name: 'galert',
   props: {
-    message: {
+    message: { // alternatively, use message slot
       type: String
     },
-    detailedMessage: {
+    type: {
+      type: String,
+      default: 'error'
+    },
+    identifier: { // pass identifier to permanently hide the message on close
       type: String
     },
     color: {
-      type: String,
-      required: true
-    },
-    tile: {
-      type: Boolean
+      type: String
     },
-    dark: {
-      type: Boolean,
-      default: true
+    transition: {
+      type: String
     }
   },
   data () {
     return {
-      detailedMessageVisible: false
+      alertVisible: false
     }
   },
   computed: {
-    alertVisible: {
-      get () {
-        return !!this.message
-      },
-      set (value) {
-        if (!value) {
-          this.$emit('update:message', undefined)
-          this.$emit('update:detailedMessage', undefined)
-        }
+    messageHtml () {
+      return transformHtml(this.message)
+    }
+  },
+  methods: {
+    ...mapActions([
+      'setAlertBanner'
+    ]),
+    async closeBanner () {
+      if (this.identifier) { // hide permanently
+        this.hidePermanently(this.identifier)
       }
+
+      this.setAlertVisibility(false)
+    },
+    hidePermanently (identifier) {
+      const permanentlyHiddenIds = this.$localStorage.getObject(LOCAL_STORE_ALERT_BANNER_HIDDEN_MESSAGES) || {}
```

Comment:
Why don't we write these values into the vuex store and decide there centrally (via localStorage plugin) what to persist in localStroage. I think this would be much cleaner.

Reply:
```js
import debounce from 'lodash/debounce'

const ALERT_BANNER_HIDDEN_MESSAGES = 'global/alert-banner/hidden-messages'
const SET_ALERT_BANNER_HIDDEN_MESSAGES = 'SET_ALERT_BANNER_HIDDEN_MESSAGES'


const setAlertBannerHiddenMessages = debounce(value => {
  localStorage.setItem(ALERT_BANNER_HIDDEN_MESSAGES, JSON.stringify(value))
}, 200)

const LocalStoragePlugin= store => {
  const value = localStorage.getItem(ALERT_BANNER_HIDDEN_MESSAGES)
  if (value) {
    store.commit(SET_ALERT_BANNER_HIDDEN_MESSAGES, JSON.parse(value))
  }
  store.subscribe((mutation, state) => {
    switch (mutation.type) {
      case SET_ALERT_BANNER_HIDDEN_MESSAGES: 
        setAlertBannerHiddenMessages(mutation.payload)
        break
    }
  })
}

const store = new Vuex.Store({
  state,
  getters,
  actions,
  mutations,
  plugins: [LocalStoragePlugin]
})
```

Reply:
Let's create a separate ticket for this.

Reply:
https://github.com/gardener/dashboard/issues/905